aten/src/ATen/native/cuda/RangeFactories.cu

```diff
@@ -6,67 +6,43 @@
 #include <cmath>
 #include <limits>
 
-#include <thrust/device_ptr.h>
-#include <thrust/sequence.h>
-#include <thrust/execution_policy.h>
+#include <ATen/native/cuda/Loops.cuh>
 
 namespace at {
 namespace native {
 
-template<typename T, typename accT = T>
-struct LinspaceOp {
-  __host__ __device__ LinspaceOp(accT start, accT step):
-    start_(start), step_(step) { }
-  __device__ __forceinline__ T operator()(ptrdiff_t index) {
-    accT increment = step_ * static_cast<accT>(index);
-    accT value = start_ + increment;
-    return static_cast<T>(value);
-  }
-
-  const accT start_, step_;
-};
-
-template<typename T, typename accT = T>
-struct LogspaceOp {
-  __host__ __device__ LogspaceOp(accT start, accT step, accT base):
-    start_(start), step_(step), base_(base) { }
-  __device__ __forceinline__ T operator()(ptrdiff_t index) {
-    accT increment = step_ * static_cast<accT>(index);
-    accT value = std::pow(base_, start_ + increment);
-    return static_cast<T>(value);
-  }
-
-  const accT start_, step_, base_;
-};
-
 Tensor& linspace_cuda_out(Tensor& result, Scalar start, Scalar end, int64_t steps) {
   TORCH_CHECK(steps >= 0, "number of steps must be non-negative");
 
   if (result.numel() != steps) {
     result.resize_({steps});
   }
-  Tensor r = result.is_contiguous() ? result : result.contiguous();
+  // Using TensorIter, output no longer need to be contiguous
+  // In the case of not doing resize_(), need to check whether output have internal overlap
+  TORCH_CHECK(has_internal_overlap(result) == MemOverlap::NO,
```

Comment:
can you do the same workaround for possibly overlapping result as before (that is, make contiguous intermediate tensor, and copy it result)? copy_ will work for TOO_HARD case (even though it's potentially unsafe)